pandas/tests/indexes/period/test_indexing.py

```diff
@@ -685,3 +685,44 @@ def test_period_index_indexer(self):
         tm.assert_frame_equal(df, df.loc[list(idx)])
         tm.assert_frame_equal(df.iloc[0:5], df.loc[idx[0:5]])
         tm.assert_frame_equal(df, df.loc[list(idx)])
+
+    @pytest.mark.parametrize(
+        "key, err ,msg",
+        [
+            (
+                PeriodIndex(["2016"], freq="A"),
+                TypeError,
+                (
+                    r"'PeriodIndex\(\['2016'\], dtype='period\[A-DEC\]', "
+                    r"freq='A-DEC'\)'"
+                    r" is an invalid key"
+                ),
+            ),
+            (2015, KeyError, "2015"),
+            ("foo-bar", KeyError, "Cannot interpret 'foo-bar' as period"),
+            ((2015, 2016), TypeError, r"'\(2015, 2016\)' is an invalid key"),
```

Comment:
it looks like the main difference is that this raises KeyError in master.  am i missing something else?

Reply:
maybe TypeError is more accurate/useful?